### ISSUE TITLE
Test 'renv' method of installing R from repo.gsc.wustl.edu

### ISSUE THREAD

**Issue by malachig** [genome/gms#126]
To test installation of R by new method developed by @kindjal 

Add this repo to your APT sources (/etc/apt/sources.list.d/genome.list):
`deb http://repo.gsc.wustl.edu/dev/ubuntu/ precise-genome-systems main`
`deb http://repo.gsc.wustl.edu/dev/ubuntu/ precise-genome-development main`
`sudo apt-get update`

Install R2.15.3 (2.15.3-1precise1~20140305) as follows:
`sudo apt-cache policy renv`
`sudo apt-get install renv`

After installation, because it has an /etc/profile.d/renv.sh file that modifies PATH, you must **logout** and log in again

Then test environment as follows:

```
echo $PATH
#/home/vagrant/bin:/opt/gms/DDWJC25/sw/apps/R/R-2.15.2/bin_safe:/opt/gms/DDWJC25/sw/ur/bin:/opt/gms/DDWJC25/sw/workflow/bin:/opt/gms/DDWJC25/sw/genome/bin:/opt/gms/DDWJC25/sw/legacy/bin:/opt/openlava-2.2/bin:/opt/renv/.renv/shims:/opt/renv/.renv/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/opt/vagrant_ruby/bin
```

```
which Renv
#/opt/renv/.renv/bin/Renv
```

```
Renv versions
#  2.15.3
```

Set Renv version to use and launch R

```
Renv shell 2.15.3
Renv versions
which R
/opt/renv/.renv/shims/R
```

To set an Renv persistently, do the following:
`sudo -i -u root Renv global 2.15.3`

**Comment by mcallaway** [comment]
This is ready for another test:

```
sudo apt-get update
sudo apt-get install --reinstall renv
vagrant@precise2:~$ sudo apt-get install --reinstall renv -y
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following extra packages will be installed:
  fontconfig fontconfig-config libcairo2 libdatrie1 libfontconfig1 libfontenc1 libgfortran3 libgl1-mesa-dri
  libgl1-mesa-glx libglapi-mesa libhdf5-serial-1.8.4 libice6 libicu48 libjpeg-turbo8 libjpeg8 libllvm3.0 libpango1.0-0
  libpixman-1-0 libsm6 libthai-data libthai0 libtiff4 libutempter0 libx11-xcb1 libxaw7 libxcb-glx0 libxcb-render0
  libxcb-shape0 libxcb-shm0 libxcomposite1 libxdamage1 libxfixes3 libxft2 libxi6 libxinerama1 libxmu6 libxpm4
  libxrender1 libxss1 libxt6 libxtst6 libxv1 libxxf86dga1 libxxf86vm1 tk8.5 ttf-dejavu-core x11-common x11-utils
  xbitmaps xterm
Suggested packages:
  libglide3 ttf-baekmuk ttf-arphic-gbsn00lp ttf-arphic-bsmi00lp ttf-arphic-gkai00mp ttf-arphic-bkai00mp mesa-utils
  xfonts-cyrillic
The following NEW packages will be installed:
  fontconfig fontconfig-config libcairo2 libdatrie1 libfontconfig1 libfontenc1 libgfortran3 libgl1-mesa-dri
  libgl1-mesa-glx libglapi-mesa libhdf5-serial-1.8.4 libice6 libicu48 libjpeg-turbo8 libjpeg8 libllvm3.0 libpango1.0-0
  libpixman-1-0 libsm6 libthai-data libthai0 libtiff4 libutempter0 libx11-xcb1 libxaw7 libxcb-glx0 libxcb-render0
  libxcb-shape0 libxcb-shm0 libxcomposite1 libxdamage1 libxfixes3 libxft2 libxi6 libxinerama1 libxmu6 libxpm4
  libxrender1 libxss1 libxt6 libxtst6 libxv1 libxxf86dga1 libxxf86vm1 tk8.5 ttf-dejavu-core x11-common x11-utils
  xbitmaps xterm
The following packages will be upgraded:
  renv
...
```

**Comment by mcallaway** [comment]
```
vagrant@precise2:~$ Renv version
2.15.3 (set by /opt/renv/.renv/version)
vagrant@precise2:~$ R

R version 2.15.3 (2013-03-01) -- "Security Blanket"
Copyright (C) 2013 The R Foundation for Statistical Computing
ISBN 3-900051-07-0
Platform: x86_64-unknown-linux-gnu (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

  Natural language support but running in an English locale

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

> capabilities()
    jpeg      png     tiff    tcltk      X11     aqua http/ftp  sockets
    TRUE     TRUE     TRUE     TRUE    FALSE    FALSE     TRUE     TRUE
  libxml     fifo   cledit    iconv      NLS  profmem    cairo
    TRUE     TRUE     TRUE     TRUE     TRUE     TRUE     TRUE
```

**Comment by malachig** [comment]
It looks like we now have an R installation via an renv debian package that has all the capabilities and CRAN and Bioconductor packages used by the GMS.  It is very similar to the custom R installation we are using now except it has ~20 additional packages (from auto grabbing of dependencies) and some package versions are newer...

Next step for testing is to add it to genome-snapshot-deps as follows: renv >= 2.15.3-1precise1~20140313

From within the firewall we can also get it directly from the APT server.  If our internal APT server gets published, we could eventually just hit the public APT server as well...

**Comment by malachig** [comment]
From @kindjal. The TGI public APT repos look like this:

```
# lucid_genome
deb http://apt.genome.wustl.edu/ubuntu lucid-genome main non-free
# precise_genome
deb http://apt.genome.wustl.edu/ubuntu precise-genome main non-free
# precise_genome_development
Explanation: : precise_genome
Package: *
Pin: origin apt.genome.wustl.edu
Pin-Priority: 1001
# precise_genome_development
Explanation: : lucid_genome
Package: *
Pin: origin apt.genome.wustl.edu
Pin-Priority: 1001
```

**Comment by gatoravi** [comment]
this is being tested in the APT dump version 'renv_test'. To test this, following changes were made to the Makefile,

``` bash
-APT_DUMP_VERSION=2014.03.17
+APT_DUMP_VERSION=renv_test


-sudo bash -l -c 'source /etc/genome.conf; make done-host/rails done-host/apache done-host/db-schema done-host/openlava-install done-host/custom-r done-host/exim-config'
+sudo bash -l -c 'source /etc/genome.conf; make done-host/rails done-host/apache done-host/db-schema done-host/openlava-install done-host/exim-config'

-sudo apt-get install -q -y --force-yes git-core vim byobu nfs-common perl-doc genome-snapshot-deps `cat setup/packages.lst` `cat setup/install_custom_r/packages.r.lst`
+sudo apt-get install -q -y --force-yes git-core vim byobu nfs-common perl-doc genome-snapshot-deps `cat setup/packages.lst`
```

**Comment by gatoravi** [comment]
This apt-dump installed cleanly without any errors. When R is loaded after the install, R-2.15.3 is loaded by default.

``` bash
vagrant@precise64 /vagrant/setup/install_custom_r (ubuntu-12.04)> which R
/opt/renv/.renv/shims/R
```

Running `/usr/bin/R` loads R-3.0.3, 

``` bash
vagrant@precise64 /vagrant/setup/install_custom_r (ubuntu-12.04)> /usr/bin/R

R version 3.0.3 (2014-03-06) -- "Warm Puppy"
```

Running `R CMD BATCH install_r_packages.R install_r_packages.Rout` (this script is in the gms repo in gms/setup/install_custom_r) seems to find no errors. All the packages seem to load successfully. The next step might be to test builds with this installation, either on clia1 or the blade.

**Comment by gatoravi** [comment]
The makefile has been updated to use the renv method of install. The custom R installation has been removed from the Makefile. This installation was tested on clia1 and builds for all models succeeded(except for genotype microarray models which are still running !!). The new apt-repo and R installation changes have been pushed here - https://github.com/genome/gms/commit/f6072f58f4b75c1674ac304f2982c6b540a46954 
This cuts down the install time by quite a bit.

**Comment by malachig** [comment]
I am leaving this issue open until we have carefully reviewed graphical results and other result files generated with R in rna-seq and clin-seq builds.  For example:
- Tophat junction splicing results in rna-seq
- mutation-spectrum plots in clin-seq
- clonality plots in clin-seq
- SNV WGS/Exome/RNA integration plots in clin-seq
- Differential expression results in clin-seq
- etc.

**Comment by malachig** [comment]
I have reviewed all of these results on the latest clia1 test run using the complete HCC1395 data. Everything seems to be in order.  The Renv installation of R seems to work nicely, as far as we can tell has all the capabilities of the R install used within TGI and install much faster than the custom build we were doing before.  Many thanks to @kindjal for helping us sort all of this out.

Closing this issue now.
